src/query/service/src/interpreters/interpreter.rs

```diff
@@ -289,6 +290,8 @@ pub fn on_execution_finished(info: &ExecutionInfo, query_ctx: Arc<QueryContext>)
 
     hook_vacuum_temp_files(&query_ctx)?;
 
+    hook_disk_temp_dir(&query_ctx)?;
```

Comment:
TODO: We can use RAII to track the creation and deletion of each temporary file, which can significantly improve disk space utilization